src/bin/server.rs

```diff
@@ -56,7 +59,32 @@ fn main() {
 
     let server = if dotenv::var("USE_HYPER").is_ok() {
         println!("Booting with a hyper based server");
-        Hyper(HyperService::new(app, threads as usize))
+        let addr = ([127, 0, 0, 1], port).into();
+        let service = HyperService::new(app, threads as usize);
+        let server = hyper::Server::bind(&addr).serve(service);
+
+        let (tx, rx) = futures::sync::oneshot::channel::<()>();
+        let server = server
+            .with_graceful_shutdown(rx)
+            .map_err(|e| log::error!("Server error: {}", e));
+
+        ctrlc_handler(move || tx.send(()).unwrap_or(()));
+
+        let mut rt = tokio::runtime::Builder::new()
+            .core_threads(4)
```

Comment:
Should this be configured with an environment variable?

Reply:
I think it's fine to leave this hard-coded for now.  I added it mainly because sometimes we see the server boot with different levels of initial memory consumption and I'm trying to reduce potential variability (although I doubt this will really affect anything).  Long term, I expect we can switch back to all the defaults provided by `Runtime::new()`, which will set the number of core threads to the CPU count.